lux-lib/src/build/utils.rs

```diff
@@ -237,39 +299,70 @@ pub(crate) fn compile_c_modules(
         build.define(name, value.as_deref());
     }
 
-    let file = target.file_name().unwrap_or_else(|| {
-        panic!(
-            "Couldn't determine filename for path {}",
-            target.to_str().unwrap_or("")
-        )
-    });
+    let file = target
+        .file_name()
+        .expect("Couldn't determine filename")
+        .to_string_lossy()
+        .to_string();
     // See https://github.com/rust-lang/cc-rs/issues/594#issuecomment-2110551057
-    let objects = build.try_compile_intermediates()?;
+    let objects = build
+        .try_compile_intermediates()
+        .map_err(BuildError::CompileIntermediatesError)?;
+
+    let libdir_args = data.libdirs.iter().map(|libdir| {
+        if is_msvc {
+            format!("/LIBPATH:{}", source_dir.join(libdir).display())
+        } else {
+            format!("-L{}", source_dir.join(libdir).display())
+        }
+    });
 
-    let libdir_args = data
-        .libdirs
-        .iter()
-        .map(|libdir| format!("-L{}", source_dir.join(libdir).to_str().unwrap()));
+    let library_args = data.libraries.iter().map(|library| {
+        if is_msvc {
+            format!("{}.lib", library.to_str().unwrap())
+        } else {
+            format!("-l{}", library.to_str().unwrap())
+        }
+    });
 
-    let library_args = data
-        .libraries
-        .iter()
-        .map(|library| format!("-l{}", library.to_str().unwrap()));
-
-    let output = build
-        .try_get_compiler()?
-        .to_command()
-        .args(["-shared", "-o"])
-        .arg(parent.join(file))
-        .arg(format!("-L{}", lua.lib_dir.to_string_lossy())) // TODO: In luarocks, this is behind a link_lua_explicitly config option Library directory
-        .args(lua.link_args())
-        .args(&objects)
-        .args(libdir_args)
-        .args(library_args)
-        .output()?;
+    let output_path = parent.join(&file);
+    let output = if is_msvc {
+        let def_temp_dir = tempdir::TempDir::new("msvc-def")?.into_path().to_path_buf();
+        let def_file = mk_def_file(def_temp_dir, &file, target_module)?;
+        build
+            .try_get_compiler()?
+            .to_command()
+            .arg("/nologo")
```

Comment:
question: I see refactors from `/nologo` to `/NOLOGO`, should this also be changed?